resources/measures/advanced_rtu_control/measure.rb

```diff
@@ -0,0 +1,335 @@
+						component.setSupplyAirFlowRateDuringCoolingOperation(max_flow*des_supply_airflow) #Set max based on limit after retrofit 
+						component.setSupplyAirFlowRateDuringHeatingOperation(max_flow*des_supply_airflow) #Set max based on limit after retrofit 
+						component.setSupplyAirFlowRateWhenNoCoolingorHeatingisRequired(min_flow*des_supply_airflow) #Set min based on limit after retrofit 
+					elsif air_loop_hvac.designSupplyAirFlowRate.is_initialized
+						des_supply_airflow = air_loop_hvac.designSupplyAirFlowRate.get
+						#puts ("des supply airflow" + "#{air_loop_hvac.name.to_s}" "#{des_supply_airflow}" + "new max" + "#{max_flow*des_supply_airflow}")
+						component.setSupplyAirFlowRateDuringCoolingOperation(max_flow*des_supply_airflow)
+						component.setSupplyAirFlowRateDuringHeatingOperation(max_flow*des_supply_airflow)
+						component.setSupplyAirFlowRateWhenNoCoolingorHeatingisRequired(min_flow*des_supply_airflow)
+					end 
+				sup_fan = component.supplyFan
+				if sup_fan.is_initialized
+					sup_fan = sup_fan.get
+					#handle fan on off objects; replace FanOnOff with FanVariableVolume 
+                    if sup_fan.to_FanOnOff.is_initialized
+					   puts "fan on off" 
+					   sup_fan = sup_fan.to_FanOnOff.get
+					   pressure_rise = sup_fan.pressureRise()
+					   #Create VS supply fan 
+					   fan = OpenStudio::Model::FanVariableVolume.new(model)
+					   fan.setName("#{air_loop_hvac.name} Fan")
+					   fan.setFanEfficiency(fan_tot_eff) 
+					   fan.setFanPowerMinimumFlowRateInputMethod("Fraction")
+					   fan.setPressureRise(pressure_rise)#keep it the same as the existing fan, since the balance of systems is the same 
+					   fan.setFanPowerMinimumFlowFraction(min_flow_fraction)
+					   fan.setMotorEfficiency(fan_mot_eff) unless fan_mot_eff.nil?
+					   #Add it to the unitary sys
+					   #component.resetSupplyFan()
+					   component.setSupplyFan(fan) 
+					end 
+					#handle constant speed fan objects; replace FanConstantVolume with FanVariableVolume 
+					if sup_fan.to_FanConstantVolume.is_initialized
+					   sup_fan = sup_fan.to_FanConstantVolume.get
+					   pressure_rise = sup_fan.pressureRise()
+					   #Create VS supply fan 
+					   fan = OpenStudio::Model::FanVariableVolume.new(model)
+					   fan.setName("#{air_loop_hvac.name} Fan")
+					   fan.setFanEfficiency(fan_tot_eff) 
+					   fan.setFanPowerMinimumFlowRateInputMethod("Fraction")
+					   fan.setPressureRise(pressure_rise)#keep it the same as the existing fan, since the balance of systems is the same 
+					   fan.setFanPowerMinimumFlowFraction(min_flow_fraction)
+					   fan.setMotorEfficiency(fan_mot_eff) unless fan_mot_eff.nil?
+					   #Add it to the unitary sys
+					   #component.resetSupplyFan()
+					   component.setSupplyFan(fan) 
+					end 
+				end 
+			 end 
+			 end 
+			 air_loop_hvac.thermalZones.each do |thermal_zone|
```

Comment:
For getting the min OA flow rate, you could first check to see if the "Minimum Outdoor Air Flow Rate" field is initialized and non zero. If not, then you can proceed to use manual methods for determining design OA flow rate.

Reply:
Thanks--done!